### PR TITLE
feat(useElementHover): directive support

### PR DESCRIPTION
**Pull request by webfansplz**
### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useElementHover/directive.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    },
+  },
+
+  template: `<template>
+  <div v-hover="onHover">Hover me</div>
+  </template>
+  `,
+})
+
+describe('vOnLongPress', () => {
```
Comment (okxiaoliang4) [inline packages/core/useElementHover/directive.test.ts:21]: typo

Comment (webfansplz) [reply packages/core/useElementHover/directive.test.ts]: Thanks !

File: packages/core/useElementHover/index.md [hunk 2]
```diff
@@ -22,3 +24,25 @@ const myHoverableElement = ref()
...
+import { vHover } from '@vueuse/components'
+
+const isHovered = ref(false)
+function onHover(state: boolean) {
+  isHovered.value = state
+}
+</script>
+
+<template>
+  <button v-hover="onHover">
```
Comment (antfu) [inline packages/core/useElementHover/index.md:43]: I think we should match with the function name and call it vElementHover

File: packages/core/useElementHover/directive.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+type BindingValueFunction = (state: boolean) => void
+
+export const vHover: FunctionDirective<
+HTMLElement,
+BindingValueFunction
+> = (el, binding) => {
+  if (typeof binding.value === 'function') {
+    const isHovered = useElementHover(el)
+    watch(isHovered, v => binding.value(v))
```
Comment (okxiaoliang4) [inline packages/core/useElementHover/directive.ts:13]: I think replace `watchEffect` has better performance.
```suggestion
    watchEffect(() => binding.value(unref(isHovered)))
```

File: packages/core/useElementHover/index.md (hunk 2 above)
Comment (webfansplz) [reply packages/core/useElementHover/index.md]: Fixed

File: packages/core/useElementHover/directive.ts (hunk 3 above)
Comment (webfansplz) [reply packages/core/useElementHover/directive.ts:13]: The watchEffect will triggered once immediately which will give The user unexpected behavior . (triggered `onHover` callback)

**Comment by antfu** [comment]
Ouch, please reopen it against main 😅

File: packages/core/useElementHover/directive.ts (hunk 3 above)
Comment (okxiaoliang4) [reply packages/core/useElementHover/directive.ts:13]: but it looks like when `isHovered` change then trigger the callback and pass `isHovered`.

Comment (okxiaoliang4) [reply packages/core/useElementHover/directive.ts:13]: maybe you want like this?
```typescript
if(isHovered) binding.value(unref(isHovered))
```

Comment (webfansplz) [reply packages/core/useElementHover/directive.ts:13]: I think we also need triggered the callback  while the state is `false`

**Comment by flybbjcs70** [comment]
it seems useEventListener ```tryOnScopeDispose(stop)``` cannot work in  directive  mount hook

**Comment by flybbjcs70** [comment]
```js
binding.instance.scope.run(()=>{
})
```
This may be better?

**Pull request closed without merging**